internal/prober/multihttp/script.go

```diff
@@ -17,13 +18,37 @@ import (
 //go:embed script.tmpl
 var templateFS embed.FS
 
-func buildUrl(req *sm.MultiHttpEntryRequest) string {
-	// If we are here, the request has already been validated, and the URL
-	// should be valid. This function should never return an error.
-	u, _ := url.Parse(req.Url)
+func replaceVariablesInString(s string) string {
+	re := regexp.MustCompile(`\$\{(.*?)\}`)
+	variableNames := re.FindAll([]byte(s), -1)
+	transformed := s
+	if len(variableNames) > 0 {
+		for _, variableName := range variableNames {
+			variableString := string(variableName)
+			strippedVarName := strings.Replace(variableString, "${", "", 1)
+			strippedVarName = strings.Replace(strippedVarName, "}", "", 1)
```

Comment:
the capture is the variable name, so if the input is `foo${bar}baz`, the capture is `bar`.

```
strings.ReplaceAll(input, "${" + name + "}", `${vars[` + name + `]}`)
```

should do the trick, no?

Reply:
I don't understand why, but the capture is is actually `${bar}` right now